response_operations_ui/views/collection_exercise.py

```diff
@@ -326,23 +326,40 @@ def get_create_collection_exercise_form(survey_ref, short_name):
     survey_details = survey_controllers.get_survey(short_name)
     return render_template('create-collection-exercise.html', form=form, short_name=short_name,
                            survey_ref=survey_ref, survey_id=survey_details['survey']['id'],
-                           survey_name=survey_details['collection_exercises'][0]['name'])
+                           survey_name=survey_details['survey']['shortName'])
 
 
 @collection_exercise_bp.route('/<survey_ref>-<short_name>/create-collection-exercise', methods=['POST'])
 @login_required
 def create_collection_exercise(survey_ref, short_name):
-    form = CreateCollectionExerciseDetailsForm(form=request.form)
-    if not form.validate():
+    ce_form = CreateCollectionExerciseDetailsForm(form=request.form)
+
+    survey_details = survey_controllers.get_survey(short_name)
+
+    if not ce_form.validate():
         logger.info("Failed validation, retrieving survey data for form", short_name=short_name, survey_ref=survey_ref)
-        survey_details = survey_controllers.get_survey(short_name)
-        return render_template('create-collection-exercise.html', form=form, short_name=short_name, errors=form.errors,
+        error = ce_form.errors['period'][1] \
+            if ce_form.errors['period'][1] == 'Please enter numbers only for the period' else None
+        return render_template('create-collection-exercise.html', form=ce_form, short_name=short_name, errors=error,
                                survey_ref=survey_ref, survey_id=survey_details['survey']['id'],
-                               survey_name=survey_details['collection_exercises'][0]['name'])
+                               survey_name=survey_details['survey']['shortName'])
 
     else:
         logger.info("Creating collection exercise for survey", short_name=short_name, survey_ref=survey_ref)
         form = request.form
+
+        created_period = form.get('period')
+        ce_details = collection_exercise_controllers.get_collection_exercises_by_survey(survey_details['survey']['id'])
+
+        if ce_details:
```

Comment:
Will not need this check if get_collection_exercises_by_survey returns an empty list.